File: week7/count_word.py
#!/usr/bin/python3

import sys, re

arr = list(filter(None, re.split(r'[^a-zA-Z]+',' '.join(sys.stdin.readlines()))));
word = sys.argv[1].lower()

count = len(list(filter(lambda x: x == word,list(map(lambda x: x.lower(), arr)))))
print (word + " occurred "+str(count)+" times")

File: week7/count_word.pl
#!/usr/bin/perl

use warnings;
use strict;

my $word = lc $ARGV[0];
my @arr = <STDIN>;
@arr = split(/[^a-zA-Z]+/, join(' ',@arr));

my $count = 0;
foreach my $elem (@arr) {
  chomp $elem;
  $elem = lc $elem;
  if ( $elem eq $word ) {
    $count ++;
  }
}

print "$ARGV[0] occured $count times\n";
